extensions/amp-carousel/0.1/scrollable-carousel.js

```diff
@@ -89,6 +89,11 @@ export class AmpScrollableCarousel extends BaseCarousel {
       },
       ActionTrust.LOW
     );
+    /** If the element is in an email document, allow its `goToSlide` action. */
+    Services.actionServiceForDoc(this.element).maybeAddToEmailWhitelist(
```

Comment:
What do you think about making this part of `BaseElement.registerAction`?

Reply:
Interesting suggestion. I thought briefly about this option, but the case to guard against is when a component registers an action that is not intended to be allowed in the email format. For example, `amp-carousel:0.1` registers both `goToSlide` (allowed in email) and `toggleAutoplay` (disallowed in email).

One option is to pass in an additional boolean parameter to `registerAction`, i.e. `isAllowedInEmail`. Another is for `registerAction` to both register the action and add it to the allowlist if one exists, but only register disallowed-in-email actions in non-email contexts.

Neither of these options are necessarily cleaner than the current approach IMO, but open to being convinced if you feel this or something else is better.

Reply:
> One option is to pass in an additional boolean parameter to registerAction

Yea that's what I had in mind. Optional boolean or `bannedFormats` array (see format.js) or options bag that defaults to "no" for email. 

That might be a bunch of refactoring though. A simpler option is to modify `addToWhitelist`.

> Neither of these options are necessarily cleaner than the current approach IMO

What does a new API buy us here?

Reply:
Both the new API and a modification to `addToWhitelist` help clarify intent, is all. Sounds reasonable to me to go with the modification approach. Is this particularly better for bundle size or readability, you think?

Reply:
I think modifying `addToWhitelist` with a format parameter is a more scalable API. Imagine `maybeAddToAdsWhitelist`, `maybeAddToActionsWhitelist` etc.